feat: update USB and ONVIF protocols (#1202)
edgexfoundry/edgex-docs: docs_src/microservices/device/supported/device-onvif-camera/supplementary-info/ONVIF-protocol.md

This Onvif Camera Device Service is developed to control/communicate ONVIF-compliant cameras accessible via http in an EdgeX deployment

## Table of Contents
- [API Specifications](#device-onvif-api-specifications)  
- [ONVIF Protocol Properties](#onvif-protocol-properties)  
- [ONVIF Features](#onvif-features)  
- [Custom Features](#custom-features)  
- [How does the service work?](#how-does-the-device-service-work)  
- [Tested Onvif Cameras](#tested-onvif-cameras)

## Device ONVIF API Specifications
The latest version {{version}} of the device service API specifications can be found
[here](../swagger.md).

## ONVIF Protocol Properties

| Property | Description | EdgeX Value Type |               
| -- | -- | -- |  
| Address | The IP address of the camera. | String |  
| Port | The internet port of the camera. | String |  
| EndpointRefAddress | A GUID used to indentify the camera endpoint. | String |  
| DeviceStatus | The discovery/authorization status of the camera.  | String |  
| &nbsp;&nbsp;&nbsp;&nbsp; *UpWithAuth* | The device can execute commands requiring credentials. |  |

> | &nbsp;&nbsp;&nbsp;&nbsp; *UpWithoutAuth* | The devie can only execute commands that do not require credentials. Usually this means the camera's credentials have not been registered with the service yet, or have been changed. |  |

Reviewer: ```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp; *UpWithoutAuth* | The device can only execute commands that do not require credentials. Usually this means the camera's credentials have not been registered with the service yet, or have been changed. |  |
```

This Onvif Camera Device Service is developed to control/communicate ONVIF-compliant cameras accessible via http in an EdgeX deployment

## Table of Contents
- [API Specifications](#device-onvif-api-specifications)  
- [ONVIF Protocol Properties](#onvif-protocol-properties)  
- [ONVIF Features](#onvif-features)  
- [Custom Features](#custom-features)  
- [How does the service work?](#how-does-the-device-service-work)  
- [Tested Onvif Cameras](#tested-onvif-cameras)

## Device ONVIF API Specifications
The latest version {{version}} of the device service API specifications can be found
[here](../swagger.md).

## ONVIF Device Service Protocol Properties
### ONVIF Protocol

> All properties in the `ONVIF` protocol field are defined by internal device information and some user defined information.

Reviewer 2: The protocol name is defined as `Onvif` currently, and I believe it is case-sensitive.

| Property | Description | EdgeX Value Type |               
| -- | -- | -- |  
| Address | The IP address of the camera. | String |  
| Port | The internet port of the camera. | String |  
| EndpointRefAddress | A GUID used to indentify the camera endpoint. | String |  

> | DeviceStatus | The discovery/authorization status of the camera.  | String |  

Reviewer 2: ```suggestion
| DeviceStatus | The discovery and authentication status of the camera.  | String |  
```

The latest version {{version}} of the device service API specifications can be found
[here](../swagger.md).

## ONVIF Protocol Properties

| Property | Description | EdgeX Value Type |               
| -- | -- | -- |  
| Address | The IP address of the camera. | String |  
| Port | The internet port of the camera. | String |  
| EndpointRefAddress | A GUID used to indentify the camera endpoint. | String |  

> | DeviceStatus | The discovery/authorization status of the camera.  | String |  

Reviewer: I think `DeviceStatus` section can be changed.  This description is better https://docs.edgexfoundry.org/3.0/microservices/device/supported/device-onvif-camera/supplementary-info/device-status/

| &nbsp;&nbsp;&nbsp;&nbsp; *UpWithAuth* | The device can operate with full authorization. |  |
| &nbsp;&nbsp;&nbsp;&nbsp; *UpWithoutAuth* | The device only execute commands that do not require authorization. |  |
| &nbsp;&nbsp;&nbsp;&nbsp; *Reachable* | The device has been discovered, but can not execute anny commands.|  |
| &nbsp;&nbsp;&nbsp;&nbsp; *Unreachable* | The device can not be found on the network. |  |
| LastSeen | The time that the device was last seen on the network. | String |  
| MACAddress | The MAC address of the camera. | String |  
| FriendlyName | The user defined name for the camera. | String |  
| Manufacturer | The manufacturer of the camera. | String |

> | Model | The model desciption of the camera. | String | 

Reviewer: ```suggestion
| Model | The model description of the camera. | String | 
```

| FirmwareVersion | The firmware version of the camera. Format will vary by manufacturer and model. | String |
| SerialNumber | The serial number of the camera. | String |
| HardwareId | The hardware ID of the camera. | String | 

> 

Author: CustomMetadata is weird because it is it's own protocol, that is entirely user configurable. It's not a part of ProtocolProperties, but if I were to address it separately, I would just be repeating info from the CustomMetadata page

> 

Reviewer: It is part of protocol properties as per my understanding. This page is to list all the protocol properties that the device service supports. Yes the user can add any data under it but still it belongs to Protocol properties of device onvif service.
